src/_pytest/stepwise.py

```diff
@@ -78,13 +81,15 @@ def pytest_runtest_logreport(self, report):
             if self.skip:
                 # Remove test from the failed ones (if it exists) and unset the skip option
                 # to make sure the following tests will not be skipped.
-                if report.nodeid == self.lastfailed:
```

Comment:
so it turns out pytest_runtest_logreport can get called multiple times when running under xdist